runtime/src/main/java/io/quarkus/oidc/proxy/runtime/OidcProxy.java

```diff
@@ -131,134 +131,72 @@ public void authorize(RoutingContext context) {
         LOG.debug("OidcProxy: authorize");
         MultiMap queryParams = context.queryParams();
 
-        StringBuilder codeFlowParams = new StringBuilder(168); // experimentally determined to be a good size for preventing resizing and not wasting space
+        RedirectBuilder redirect = new RedirectBuilder(oidcMetadata.getAuthorizationUri());
 
-        // response_type
-        codeFlowParams.append(OidcConstants.CODE_FLOW_RESPONSE_TYPE).append("=")
-                .append(OidcConstants.CODE_FLOW_CODE);
-        // client_id
+        // required params: client_id, state, redirect_uri
         final String clientId = getClientId(queryParams.get(OidcConstants.CLIENT_ID));
         if (clientId == null) {
             LOG.error("Client id must be provided");
             badClientRequest(context);
             return;
         }
-        codeFlowParams.append("&").append(OidcConstants.CLIENT_ID).append("=")
-                .append(OidcCommonUtils.urlEncode(clientId));
-        // scope
-        String encodedScope = encodeScope(queryParams.get(OidcConstants.TOKEN_SCOPE));
-        if (encodedScope != null) {
-            codeFlowParams.append("&").append(OidcConstants.TOKEN_SCOPE).append("=")
-                    .append(encodedScope);
-        }
 
-        // state
         final String state = queryParams.get(OidcConstants.CODE_FLOW_STATE);
```

Comment:
I propose to add it explicitly since we already work with the `state` and it is one of the required ones